lib/datadog/profiling/tag_builder.rb

```diff
@@ -41,7 +43,10 @@ def self.call(
         tags[FORM_FIELD_TAG_SERVICE] = service if service
         tags[FORM_FIELD_TAG_VERSION] = version if version
 
-        user_tags.merge(tags)
+        # Make sure everything is an utf-8 string, to avoid encoding issues in native code/libddprof/further downstream
+        user_tags.merge(tags).map do |key, value|
+          [Datadog::Core::Utils.utf8_encode(key), Datadog::Core::Utils.utf8_encode(value)]
```

Comment:
Is it more expensive to allocate these arrays then coerce them `to_h`, rather than creating a new `Hash` and iterating over each set while applying the transformation?

Reply:
Creating the new hash is probably slightly cheaper, but since this is done once a minute I wouldn't expect either to have a measurable impact on an application.